$NetBSD: patch-ao,v 1.1 2008/04/08 01:44:47 bjs Exp $

We want to make sure there's no garbage in the output buffer, otherwise
untoward stuff can happen.

--- drivers/oss/oss_driver.c.orig	2008-03-04 07:20:10.000000000 -0500
+++ drivers/oss/oss_driver.c
@@ -1042,6 +1042,7 @@ static void *io_thread (void *param)
 			pthread_mutex_lock(&driver->mutex_out);
 			memcpy(localbuf, driver->outdevbuf, 
 				driver->outdevbufsize);
+			memset(driver->outdevbuf, 0x00, driver->outdevbufsize);
 			pthread_mutex_unlock(&driver->mutex_out);
 
 			io_res = write(driver->outfd, localbuf, 
